lib/galaxy/managers/collections.py

```diff
@@ -218,6 +219,39 @@ def create_dataset_collection(self, trans, collection_type, element_identifiers=
         dataset_collection.collection_type = collection_type
         return dataset_collection
 
+    def get_converters_for_collection(self, trans, id, datatypes_registry: Registry, instance_type="history"):
+        dataset_collection_instance = self.get_dataset_collection_instance(
+            trans,
+            id=id,
+            instance_type=instance_type,
+            check_ownership=True
+        )
+        dbkeys_and_extensions = dataset_collection_instance.dataset_dbkeys_and_extensions_summary
+        suitable_converters = set()
+        first_extension = True
+        most_recent_datatype = None
+        # TODO error checking
+        for datatype in dbkeys_and_extensions[1]:
+            new_converters = {}
+            new_converters = datatypes_registry.get_converters_by_datatype(datatype)
+            set_of_new_converters = set()
+            for tgt_type, tgt_val in new_converters.items():
+                converter = (tgt_type, tgt_val)
+                set_of_new_converters.add(converter)
+            if (first_extension is True):
+                suitable_converters = set_of_new_converters
+                most_recent_datatype = datatype
+                first_extension = False
+            else:
+                suitable_converters = suitable_converters.intersection(set_of_new_converters)
+                if (suitable_converters.__len__ != 0):
```

Comment:
That is very javascript ;)

```suggestion
                if not suitable_converters:
```

Reply:
I think this needs to be if suitable_converters , though... I only want to go in there if there is something still in suitable_converters.

Reply:
Yeah, you're right ... gonna blame that on the syntax 😆 😂